<a href="https://colab.research.google.com/github/ChecoSchnaider/posts/blob/master/OPI_Analitics_SSB_Exam.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# OPI ANALYTICS
## Carpetas de investigación FGJ de la Ciudad de México



### Author: Sergio Schnaider Barbosa


##### Date: 08/10/2021



#

La Ciudad de México se ubica como la segunda entidad con mayor número de habitantes de acuerdo con los resultados del Censo de Población y Vivienda de 2020, al registrar un total de 9 millones 209 mil 944 habitantes. 



In [ ]:
#importing libraries

import chart_studio
import chart_studio.plotly as py
import copy
import glob, pylab, pandas as pd
# ipywidgets for some interactive plots
from ipywidgets.widgets import * 
import ipywidgets as widgets
from math import *
import matplotlib.pylab as plt
import matplotlib.pyplot as plt
import numpy as np
import os
from os import listdir
from os.path import isfile, join
os.getcwd()
# plotly 3D interactive graphs 
import plotly
from plotly.graph_objs import *
from plotly.graph_objs import *

from datetime import datetime

In [ ]:
path = r'C:/Users/checo/opi/'
csv_path = os.path.join(path, "carpetas_ss_junio2020.csv")
df = pd.read_csv(csv_path)
    

In [ ]:
df

ao_hechos mes_hechos         fecha_hechos  ao_inicio mes_inicio  \
0           2016.0      Enero  2016-01-31 22:16:00       2016    Febrero   
1           2016.0      Enero  2016-01-31 20:50:00       2016    Febrero   
2           2016.0    Febrero  2016-02-01 00:30:00       2016    Febrero   
3           2016.0      Enero  2016-01-31 22:00:00       2016    Febrero   
4           2015.0  Diciembre  2015-12-25 12:00:00       2016    Febrero   
...            ...        ...                  ...        ...        ...   
1030757     2020.0      Junio  2020-06-30 23:00:00       2020      Junio   
1030758     2020.0      Junio  2020-06-29 03:30:00       2020      Junio   
1030759     2019.0  Diciembre  2019-12-20 22:30:00       2020      Junio   
1030760     2020.0      Junio  2020-06-30 18:40:00       2020      Junio   
1030761     2020.0      Junio  2020-06-30 21:30:00       2020      Junio   

                fecha_inicio  \
0        2016-02-01 00:25:44   
1        2016-02-01 00:52:37   
2        2016-02-01 01:33:26   
3        2016-02-01 02:09:11   
4        2016-02-01 02:16:49   
...                      ...   
1030757  2020-06-30 23:38:59   
1030758  2020-06-30 23:39:20   
1030759  2020-06-30 23:46:02   
1030760  2020-06-30 23:47:25   
1030761  2020-06-30 23:55:46   

                                                    delito  \
0                      DAÑO EN PROPIEDAD AJENA INTENCIONAL   
1        ROBO DE VEHICULO DE SERVICIO PARTICULAR CON VI...   
2                             NARCOMENUDEO POSESION SIMPLE   
3           ROBO A TRANSEUNTE EN VIA PUBLICA CON VIOLENCIA   
4                                       DENUNCIA DE HECHOS   
...                                                    ...   
1030757     ROBO A TRANSEUNTE EN VIA PUBLICA CON VIOLENCIA   
1030758                                             FRAUDE   
1030759                      NEGACION DEL SERVICIO PUBLICO   
1030760                                           AMENAZAS   
1030761                                 VIOLENCIA FAMILIAR   

                                                  fiscalia    agencia  \
0                           INVESTIGACIÓN EN BENITO JUÁREZ       BJ-2   
1        INVESTIGACIÓN PARA LA ATENCIÓN DEL DELITO DE R...  ORIENTEII   
2        INVESTIGACIÓN PARA LA ATENCIÓN DE NIÑOS, NIÑAS...         57   
3                              INVESTIGACIÓN EN IZTAPALAPA      IZP-6   
4                           INVESTIGACIÓN EN BENITO JUÁREZ       BJ-1   
...                                                    ...        ...   
1030757  FISCALÍA DE INVESTIGACIÓN TERRITORIAL EN GUSTA...      GAM-6   
1030758                   UNIDAD DE RECEPCIÓN POR INTERNET      CEN-1   
1030759  FISCALÍA PARA LA INVESTIGACIÓN DE LOS DELITOS ...          B   
1030760  FISCALÍA DE INVESTIGACIÓN TERRITORIAL EN CUAUH...      CUH-2   
1030761  FISCALÍA DE INVESTIGACIÓN TERRITORIAL EN IZTAC...      IZC-2   

                  unidad_investigacion  \
0                               UI-3CD   
1                               UI-3CD   
2                               UI-3CD   
3                               UI-3CD   
4                               UI-3SD   
...                                ...   
1030757                         UI-1CD   
1030758  Unidad de Investigación 1 S/D   
1030759                         UI-1CD   
1030760                         UI-1SD   
1030761                         UI-1SD   

                                          categoria_delito      calle_hechos  \
0                                   DELITO DE BAJO IMPACTO  CALZ. DE TLALPAN   
1                     ROBO DE VEHÍCULO CON Y SIN VIOLENCIA          SUR 25 A   
2                                   DELITO DE BAJO IMPACTO  BATALLONES ROJOS   
3        ROBO A TRANSEUNTE EN VÍA PÚBLICA CON Y SIN VIO...           GERANIO   
4                                       HECHO NO DELICTIVO     SAN FRANCISCO   
...                                                    ...               ...   
1030757  ROBO A TRANSEUN

### Analisis exploratorio de datos

In [ ]:
n = df.nunique(axis=0)
  
print("No.of.unique values in each column :\n",n)

No.of.unique values in each column :
 ao_hechos                   69
mes_hechos                  12
fecha_hechos            338104
ao_inicio                    5
mes_inicio                  12
fecha_inicio            943893
delito                     341
fiscalia                    72
agencia                    159
unidad_investigacion       101
categoria_delito            18
calle_hechos            223015
calle_hechos2           100779
colonia_hechos            1658
alcaldia_hechos            505
longitud                618304
latitud                 627929
tempo                        0
dtype: int64


Ahora se checan cuantos valores NaN existen en nuestro dataframe

In [ ]:
sumNaN = df.isna().sum().sum()

print("Number of NaNs :\n",sumNaN)

Number of NaNs :
 1781512


Se realiza un análisise de cuantas variables unicas existen por columna y se puede notar que en la columna de 'alcaldia_hechos' muestra un valor de 505, cuando solamenten en la CDMX existen 16 alcadias.

Se procedera a filtrar los datos que contengan unicamente:

['BENITO JUAREZ', 'IZTAPALAPA', 'CUAUHTEMOC', 'TLAHUAC','IZTACALCO', 'GUSTAVO A MADERO', 'MIGUEL HIDALGO', 'TLALPAN',
'ALVARO OBREGON', 'VENUSTIANO CARRANZA', 'AZCAPOTZALCO', 'CUAJIMALPA DE MORELOS', 'COYOACAN', 'XOCHIMILCO',
'LA MAGDALENA CONTRERAS', 'MILPA ALTA']


In [ ]:
df.alcaldia_hechos.unique()

array(['BENITO JUAREZ', 'IZTAPALAPA', 'CUAUHTEMOC', 'TLAHUAC',
       'IZTACALCO', 'GUSTAVO A MADERO', 'MIGUEL HIDALGO', 'TLALPAN',
       'ALVARO OBREGON', 'VENUSTIANO CARRANZA', 'AZCAPOTZALCO',
       'CUAJIMALPA DE MORELOS', 'COYOACAN', 'XOCHIMILCO',
       'LA MAGDALENA CONTRERAS', 'MILPA ALTA', 'TLALNEPANTLA DE BAZ',
       'VERACRUZ', 'IXCATEOPAN DE CUAUHTEMOC', 'TUXPAM', 'HUIXQUILUCAN',
       'OCOYOACAC', 'ECATEPEC DE MORELOS', nan, 'SIN DATO', 'TULTEPEC',
       'ACATZINGO', 'GUELATAO DE JUAREZ', 'HERMOSILLO', 'ENSENADA',
       'IXTAPALUCA', 'NAUCALPAN DE JUAREZ', 'NEZAHUALCOYOTL', 'CHALCO',
       'TLATLAYA', 'SAN JUAN GUICHICOVI', 'ZAPOPAN',
       'SAN PEDRO IXTLAHUACA', 'NICOLAS ROMERO', 'VICTORIA', 'CULIACAN',
       'VALLE DE CHALCO SOLIDARIDAD', 'TIANGUISTENCO', 'TEHUACAN',
       'ACAPULCO DE JUAREZ', 'TEMASCALCINGO', 'CUAUTITLAN IZCALLI',
       'TIJUANA', 'CUERNAVACA', 'TENANGO DEL VALLE',
       'TULANCINGO DE BRAVO', 'CORREGIDORA', 'JUAREZ HIDALGO', 'CELAYA',
    

In [ ]:
listaAlcaldias = ['BENITO JUAREZ', 'IZTAPALAPA', 'CUAUHTEMOC', 'TLAHUAC','IZTACALCO', 'GUSTAVO A MADERO', 'MIGUEL HIDALGO', 'TLALPAN',
'ALVARO OBREGON', 'VENUSTIANO CARRANZA', 'AZCAPOTZALCO', 'CUAJIMALPA DE MORELOS', 'COYOACAN', 'XOCHIMILCO','LA MAGDALENA CONTRERAS', 'MILPA ALTA']

boolSeries = df.alcaldia_hechos.isin(listaAlcaldias)
df_Filtrado = df[boolSeries] 

df_Filtrado

ao_hechos mes_hechos         fecha_hechos  ao_inicio mes_inicio  \
0           2016.0      Enero  2016-01-31 22:16:00       2016    Febrero   
1           2016.0      Enero  2016-01-31 20:50:00       2016    Febrero   
2           2016.0    Febrero  2016-02-01 00:30:00       2016    Febrero   
3           2016.0      Enero  2016-01-31 22:00:00       2016    Febrero   
4           2015.0  Diciembre  2015-12-25 12:00:00       2016    Febrero   
...            ...        ...                  ...        ...        ...   
1030757     2020.0      Junio  2020-06-30 23:00:00       2020      Junio   
1030758     2020.0      Junio  2020-06-29 03:30:00       2020      Junio   
1030759     2019.0  Diciembre  2019-12-20 22:30:00       2020      Junio   
1030760     2020.0      Junio  2020-06-30 18:40:00       2020      Junio   
1030761     2020.0      Junio  2020-06-30 21:30:00       2020      Junio   

                fecha_inicio  \
0        2016-02-01 00:25:44   
1        2016-02-01 00:52:37   
2        2016-02-01 01:33:26   
3        2016-02-01 02:09:11   
4        2016-02-01 02:16:49   
...                      ...   
1030757  2020-06-30 23:38:59   
1030758  2020-06-30 23:39:20   
1030759  2020-06-30 23:46:02   
1030760  2020-06-30 23:47:25   
1030761  2020-06-30 23:55:46   

                                                    delito  \
0                      DAÑO EN PROPIEDAD AJENA INTENCIONAL   
1        ROBO DE VEHICULO DE SERVICIO PARTICULAR CON VI...   
2                             NARCOMENUDEO POSESION SIMPLE   
3           ROBO A TRANSEUNTE EN VIA PUBLICA CON VIOLENCIA   
4                                       DENUNCIA DE HECHOS   
...                                                    ...   
1030757     ROBO A TRANSEUNTE EN VIA PUBLICA CON VIOLENCIA   
1030758                                             FRAUDE   
1030759                      NEGACION DEL SERVICIO PUBLICO   
1030760                                           AMENAZAS   
1030761                                 VIOLENCIA FAMILIAR   

                                                  fiscalia    agencia  \
0                           INVESTIGACIÓN EN BENITO JUÁREZ       BJ-2   
1        INVESTIGACIÓN PARA LA ATENCIÓN DEL DELITO DE R...  ORIENTEII   
2        INVESTIGACIÓN PARA LA ATENCIÓN DE NIÑOS, NIÑAS...         57   
3                              INVESTIGACIÓN EN IZTAPALAPA      IZP-6   
4                           INVESTIGACIÓN EN BENITO JUÁREZ       BJ-1   
...                                                    ...        ...   
1030757  FISCALÍA DE INVESTIGACIÓN TERRITORIAL EN GUSTA...      GAM-6   
1030758                   UNIDAD DE RECEPCIÓN POR INTERNET      CEN-1   
1030759  FISCALÍA PARA LA INVESTIGACIÓN DE LOS DELITOS ...          B   
1030760  FISCALÍA DE INVESTIGACIÓN TERRITORIAL EN CUAUH...      CUH-2   
1030761  FISCALÍA DE INVESTIGACIÓN TERRITORIAL EN IZTAC...      IZC-2   

                  unidad_investigacion  \
0                               UI-3CD   
1                               UI-3CD   
2                               UI-3CD   
3                               UI-3CD   
4                               UI-3SD   
...                                ...   
1030757                         UI-1CD   
1030758  Unidad de Investigación 1 S/D   
1030759                         UI-1CD   
1030760                         UI-1SD   
1030761                         UI-1SD   

                                          categoria_delito      calle_hechos  \
0                                   DELITO DE BAJO IMPACTO  CALZ. DE TLALPAN   
1                     ROBO DE VEHÍCULO CON Y SIN VIOLENCIA          SUR 25 A   
2                                   DELITO DE BAJO IMPACTO  BATALLONES ROJOS   
3        ROBO A TRANSEUNTE EN VÍA PÚBLICA CON Y SIN VIO...           GERANIO   
4                                       HECHO NO DELICTIVO     SAN FRANCISCO   
...                                                    ...               ...   
1030757  ROBO A TRANSEUN

In [ ]:
df_Filtrado.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1020094 entries, 0 to 1030761
Data columns (total 20 columns):
ao_hechos               1019707 non-null float64
mes_hechos              1019707 non-null object
fecha_hechos            1019707 non-null object
ao_inicio               1020094 non-null int64
mes_inicio              1020094 non-null object
fecha_inicio            1020094 non-null object
delito                  1020094 non-null object
fiscalia                1020092 non-null object
agencia                 1020094 non-null object
unidad_investigacion    1019870 non-null object
categoria_delito        1020094 non-null object
calle_hechos            1017701 non-null object
calle_hechos2           404825 non-null object
colonia_hechos          989249 non-null object
alcaldia_hechos         1020094 non-null object
longitud                990232 non-null float64
latitud                 990232 non-null float64
tempo                   0 non-null float64
fecha_hechos2           102009

In [ ]:
#filtering the data with >=3 ratings 
df_Filtrado = df_Filtrado[df_Filtrado['ao_hechos'] >= 2016]

Muestra de indicadores delictivos por alcaldia de la CDMX

In [ ]:
ConteoDelito = (df_Filtrado['delito'].isin([0,1]).groupby(df_Filtrado['alcaldia_hechos']).count()
                     .astype(int).reset_index(name='CuentaDelito'))
ConteoDelito

alcaldia_hechos  CuentaDelito
0           ALVARO OBREGON         67428
1             AZCAPOTZALCO         48582
2            BENITO JUAREZ         84979
3                 COYOACAN         67096
4    CUAJIMALPA DE MORELOS         14306
5               CUAUHTEMOC        156176
6         GUSTAVO A MADERO        100561
7                IZTACALCO         42535
8               IZTAPALAPA        151824
9   LA MAGDALENA CONTRERAS         15539
10          MIGUEL HIDALGO         65908
11              MILPA ALTA          6616
12                 TLAHUAC         23608
13                 TLALPAN         58478
14     VENUSTIANO CARRANZA         58380
15              XOCHIMILCO         30914

In [ ]:
DelitoTemporada = (df_Filtrado['delito'].isin([0,1]).groupby(df_Filtrado['mes_hechos']).count()
                     .astype(int).reset_index(name='MesDelito'))
DelitoTemporada


mes_hechos  MesDelito
0        Abril      84972
1       Agosto      78120
2    Diciembre      72220
3        Enero      89842
4      Febrero      90018
5        Julio      74666
6        Junio      85094
7        Marzo      95143
8         Mayo      90298
9    Noviembre      75770
10     Octubre      80593
11  Septiembre      76194

In [ ]:
dateTimeObj = datetime.now()


In [ ]:
timestampStr = dateTimeObj.strftime("%d")
print('Current Timestamp : ', timestampStr)

Current Timestamp :  09


In [ ]:
df_Filtrado['fecha_hechos2'] = df_Filtrado['fecha_hechos'].astype(str)
df_Filtrado['DOW'] = df_Filtrado['fecha_hechos2'].str.slice(start=8, stop=10)
df_Filtrado.drop(columns=['fecha_hechos2'])
df_Filtrado['DOW']


0          31
1          31
2          01
3          31
5          01
           ..
1030757    30
1030758    29
1030759    20
1030760    30
1030761    30
Name: DOW, Length: 992930, dtype: object

In [ ]:
DelitoDia = (df_Filtrado['delito'].isin([0,1]).groupby(df_Filtrado['DOW']).count()
                     .astype(int).reset_index(name='DiaDelito2'))
DelitoDia

DOW  DiaDelito2
0   01       36376
1   02       32086
2   03       32545
3   04       33268
4   05       33558
5   06       32151
6   07       32698
7   08       32718
8   09       31842
9   10       33046
10  11       32421
11  12       32736
12  13       32509
13  14       32553
14  15       36014
15  16       32611
16  17       32917
17  18       32507
18  19       32237
19  20       33620
20  21       31544
21  22       32778
22  23       32275
23  24       31262
24  25       31536
25  26       32229
26  27       32286
27  28       32026
28  29       29300
29  30       29341
30  31       17940

In [ ]:
df_Filtrado.categoria_delito.unique()

array(['DELITO DE BAJO IMPACTO', 'ROBO DE VEHÍCULO CON Y SIN VIOLENCIA',
       'ROBO A TRANSEUNTE EN VÍA PÚBLICA CON Y SIN VIOLENCIA',
       'LESIONES DOLOSAS POR DISPARO DE ARMA DE FUEGO',
       'HECHO NO DELICTIVO', 'ROBO A NEGOCIO CON VIOLENCIA',
       'ROBO A CASA HABITACIÓN CON VIOLENCIA',
       'ROBO A PASAJERO A BORDO DE MICROBUS CON Y SIN VIOLENCIA',
       'VIOLACIÓN', 'ROBO A TRANSPORTISTA CON Y SIN VIOLENCIA',
       'ROBO A CUENTAHABIENTE SALIENDO DEL CAJERO CON VIOLENCIA',
       'ROBO A REPARTIDOR CON Y SIN VIOLENCIA', 'HOMICIDIO DOLOSO',
       'ROBO A PASAJERO A BORDO DE TAXI CON VIOLENCIA',
       'ROBO A PASAJERO A BORDO DEL METRO CON Y SIN VIOLENCIA',
       'SECUESTRO', 'PLAGIO O SECUESTRO', 'FEMINICIDIO'], dtype=object)

In [ ]:
DelitoTipo = (df_Filtrado['delito'].isin([0,1]).groupby(df_Filtrado['categoria_delito']).count()
                   .astype(int).reset_index(name='CatDelito'))
DelitoTipo.sort_values(by=['CatDelito'])

categoria_delito  CatDelito
5                                  PLAGIO O SECUESTRO          1
1                                         FEMINICIDIO          2
16                                          SECUESTRO        163
14           ROBO A TRANSPORTISTA CON Y SIN VIOLENCIA        525
10      ROBO A PASAJERO A BORDO DE TAXI CON VIOLENCIA       1012
7   ROBO A CUENTAHABIENTE SALIENDO DEL CAJERO CON ...       1933
6                ROBO A CASA HABITACIÓN CON VIOLENCIA       2532
9   ROBO A PASAJERO A BORDO DE MICROBUS CON Y SIN ...       3711
17                                          VIOLACIÓN       4295
3                                    HOMICIDIO DOLOSO       5694
4       LESIONES DOLOSAS POR DISPARO DE ARMA DE FUEGO       7090
12              ROBO A REPARTIDOR CON Y SIN VIOLENCIA      10329
11  ROBO A PASAJERO A BORDO DEL METRO CON Y SIN VI...      10401
8                        ROBO A NEGOCIO CON VIOLENCIA      17319
2                                  HECHO NO DELICTIVO      48225
15               ROBO DE VEHÍCULO CON Y SIN VIOLENCIA      55053
13  ROBO A TRANSEUNTE EN VÍA PÚBLICA CON Y SIN VIO...      59587
0                              DELITO DE BAJO IMPACTO     765058

In [ ]:
DelitoAno = (df_Filtrado['delito'].isin([0,1]).groupby(df_Filtrado['ao_hechos']).count()
                   .astype(int).reset_index(name='AnoDelito'))
DelitoAno

ao_hechos  AnoDelito
0     2016.0     192700
1     2017.0     224065
2     2018.0     250253
3     2019.0     237701
4     2020.0      88211

In [ ]:
filAlcaldia = df_Filtrado.groupby(['alcaldia_hechos', 'categoria_delito']).size()
filAlcaldia

alcaldia_hechos  categoria_delito                                    
ALVARO OBREGON   DELITO DE BAJO IMPACTO                                  53100
                 FEMINICIDIO                                                 1
                 HECHO NO DELICTIVO                                       3162
                 HOMICIDIO DOLOSO                                          421
                 LESIONES DOLOSAS POR DISPARO DE ARMA DE FUEGO             580
                                                                         ...  
XOCHIMILCO       ROBO A TRANSEUNTE EN VÍA PÚBLICA CON Y SIN VIOLENCIA     2023
                 ROBO A TRANSPORTISTA CON Y SIN VIOLENCIA                    5
                 ROBO DE VEHÍCULO CON Y SIN VIOLENCIA                     2639
                 SECUESTRO                                                   5
                 VIOLACIÓN                                                 242
Length: 257, dtype: int64

Se muestra que el 77% de los crimenes son de bajo impacto. Y el 4.8% son hechos no delictivos, el restante ~18% se distribuye entre las 16 categorias de delitos. 

### Preguntas

1. ¿Qué pruebas identificarías para asegurar la calidad de estos datos? No es necesario hacerlas, sólo describe la prueba y lo que te dice cada una.


La primer prueba para identificar es el número total de NaNs que existen en la base de datos, porque muchas veces no se pueden realizar calculos o estimaciones porque no hay suficientes datos. 

Después, se pueden verificar los diferentes tipos de variables dentro de las columnas. Esto para saber que tan extenso tiene que ser el análisis exploratorio. Por ejemplo, dentro de este ejercicio en la columna de alcaldia se encontraron 505 diferentes valores, cuando solamente queriamos analizar la Ciudad de México (con 16 alcaldias).

Si hubiera más valores cuantitativos, se podrian realizar pruebas realizando graficas simples, para ver si las variables se correlacionan entre si.

Si son valores cuantitativos, tambien poder checar máximos y mínimos en los datos para encontrar incongruencias. Por ejemplo, en este dataset se encontraron años que no coincidian con el estudio (comentando que comenzaba desde 2016), por lo que se procedió a eliminar todos los datos que no correspondian de 2016 en adelante



2. Identifica los delitos que van a la alza y a la baja en la CDMX (ten cuidado con los delitos con pocas ocurrencias).

In [ ]:
YrAlcaldia = df_Filtrado.groupby(['ao_hechos','alcaldia_hechos', 'categoria_delito']).size()
YrAlcaldia

ao_hechos  alcaldia_hechos  categoria_delito                                    
2016.0     ALVARO OBREGON   DELITO DE BAJO IMPACTO                                  9800
                            HECHO NO DELICTIVO                                       930
                            HOMICIDIO DOLOSO                                          63
                            LESIONES DOLOSAS POR DISPARO DE ARMA DE FUEGO             86
                            ROBO A CASA HABITACIÓN CON VIOLENCIA                      26
                                                                                    ... 
2020.0     XOCHIMILCO       ROBO A PASAJERO A BORDO DE TAXI CON VIOLENCIA              4
                            ROBO A REPARTIDOR CON Y SIN VIOLENCIA                     25
                            ROBO A TRANSEUNTE EN VÍA PÚBLICA CON Y SIN VIOLENCIA     130
                            ROBO DE VEHÍCULO CON Y SIN VIOLENCIA                     261
                            V

3. ¿Cuál es la alcaldía que más delitos tiene y cuál es la que menos?¿Por qué crees que sea esto?

In [ ]:
ConteoDelito.sort_values(by=['CuentaDelito'])

alcaldia_hechos  CuentaDelito
11              MILPA ALTA          6616
4    CUAJIMALPA DE MORELOS         14306
9   LA MAGDALENA CONTRERAS         15539
12                 TLAHUAC         23608
15              XOCHIMILCO         30914
7                IZTACALCO         42535
1             AZCAPOTZALCO         48582
14     VENUSTIANO CARRANZA         58380
13                 TLALPAN         58478
10          MIGUEL HIDALGO         65908
3                 COYOACAN         67096
0           ALVARO OBREGON         67428
2            BENITO JUAREZ         84979
6         GUSTAVO A MADERO        100561
8               IZTAPALAPA        151824
5               CUAUHTEMOC        156176

Según la información proporcionada, la alcaldia que tiene un menor número  de crímenes es Milpa Alta con el 0.64%, mientras que la alcaldia que tiene el mayor número de incidencia delictiva es la Cuauhtémoc con 15.3% del total. 

Se puede inferir que a partir de estos datos y tomando en cuenta los indicadores económicos de la Ciudad de México la Delegación Cuauhtémoc está en el tope de la incidencia delictiva por que cuenta con una participación local del producto interno bruto (PIB) de 30.3%, mientras que Milpa Alta solo participa con el 0.05% del PIB local.

Además de esto, la alcaldía de Milpa Alta tiene un número muy bajo en rutas de transporte público. En contraste, la Cuauhtémoc cuenta todas las formas de transporte público por su ubicación. 


4. ¿Existe alguna tendencia estacional en la ocurrencia de delitos (mes, semana, día de la semana, quincenas) en la CDMX?¿A qué crees que se deba?




In [ ]:
DelitoTemporada.sort_values(by=['MesDelito'])

mes_hechos  MesDelito
2    Diciembre      72220
5        Julio      74666
9    Noviembre      75770
11  Septiembre      76194
1       Agosto      78120
10     Octubre      80593
0        Abril      84972
6        Junio      85094
3        Enero      89842
4      Febrero      90018
8         Mayo      90298
7        Marzo      95143

Según la información obtenida, los meses de julio y diciembre son los que tienen menor incidencia delictiva, esto se puede inferir a que es temporada vacacional. Mientras el mes de marzo es donde se cometen mas crimenes, esto se puede deducir por que es el final del invierno. 

In [ ]:
DelitoDia

DOW  DiaDelito2
0   01       36376
1   02       32086
2   03       32545
3   04       33268
4   05       33558
5   06       32151
6   07       32698
7   08       32718
8   09       31842
9   10       33046
10  11       32421
11  12       32736
12  13       32509
13  14       32553
14  15       36014
15  16       32611
16  17       32917
17  18       32507
18  19       32237
19  20       33620
20  21       31544
21  22       32778
22  23       32275
23  24       31262
24  25       31536
25  26       32229
26  27       32286
27  28       32026
28  29       29300
29  30       29341
30  31       17940

El resultado de arriba muestra que los dias 1eros y 15 son los que tienen un mayor numero de incidencia delictiva, esto por que normalmente son días que las personas reciben su salario.

5. ¿Cuáles son los delitos que más caracterizan a cada alcaldía? Es decir, delitos que suceden con mayor frecuencia en una alcaldía y con menor frecuencia en las demás. 

In [ ]:
ya la tengo con el filtro

6. Diseña unindicador que mida el nivel de “inseguridad”. Genéralo al nivel de desagregación que te parezca más adecuado (ej. manzana, calle, AGEB, etc.). Analiza los resultados ¿Encontraste algún patrón interesante? ¿Qué decisiones se podrían tomar con el indicador?

indicador de inseguridad


In [ ]:
filAlcaldia = df_Filtrado.groupby(['alcaldia_hechos','colonia_hechos', 'categoria_delito']).size()
filAlcaldia

alcaldia_hechos  colonia_hechos              categoria_delito                                    
ALVARO OBREGON   19 DE MAYO                  DELITO DE BAJO IMPACTO                                  39
                                             HOMICIDIO DOLOSO                                         1
                                             LESIONES DOLOSAS POR DISPARO DE ARMA DE FUEGO            1
                                             ROBO A NEGOCIO CON VIOLENCIA                             1
                                             ROBO A REPARTIDOR CON Y SIN VIOLENCIA                    4
                                                                                                     ..
XOCHIMILCO       ZACATEPEC(SAN MATEO XALPA)  ROBO A CASA HABITACIÓN CON VIOLENCIA                     1
                                             ROBO A REPARTIDOR CON Y SIN VIOLENCIA                    1
                                             ROBO A TRANSEUNTE EN VÍA 